vnext/JSI/Shared/ChakraObjectRef.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "jsi/jsi.h"
+
+#ifdef CHAKRACORE
+#include "ChakraCore.h"
+#else
+#ifndef USE_EDGEMODE_JSRT
+#define USE_EDGEMODE_JSRT
+#endif
+#include <jsrt.h>
+#endif
+
+#include <exception>
+#include <string>
+
+namespace Microsoft::JSI {
+
+inline void CrashUponChakraError(JsErrorCode error) {
+  if (error != JsNoError) {
+    std::terminate();
+  }
+}
+
+void ThrowUponChakraError(
```

Comment:
>ThrowUponChakraError [](start = 5, length = 20)

Why not make Api name VerifyJsErrorElseThrow?

Reply:
This function is called in ThrowUponJsError (defined in ChakraRuntime.cpp), which does some extra things for the case where the JS bundle throws a JS exception. I'll rename this function to be VerifyChakraErrorElseThrow and rename ThrowUponJsError to VerifyJsErrorElseThrow.

---
In reply to: [336153074](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336153074) [](ancestors = 336153074)